controllers/admin/AdminCartRulesController.php

```diff
@@ -530,6 +526,21 @@ public function getProductRuleDisplay($product_rule_group_id, $product_rule_id,
         return $this->createTemplate('product_rule.tpl')->fetch();
     }
 
+    public function populateCategories(&$flatCategories, $currentCategoryTree, $currentPath = '')
+    {
+        $separator = ' > ';
+        if ($currentCategoryTree) {
```

Comment:
```suggestion
        if ($currentCategoryTree) {
            return;
        }
        
        $separator = ' > ';
```

Make code easier by don't declare useless variable when it's not needed, and return.

Also, I'm not a huge fan of using references in parameters when it can be avoided.